editoast/editoast_models/src/lib.rs

```diff
@@ -1,16 +1,23 @@
-use diesel_async::{pooled_connection::deadpool::Pool, AsyncPgConnection};
+use db_connection_pool::DatabasePoolError;
+use diesel_async::pooled_connection::deadpool::Pool;
+use diesel_async::AsyncPgConnection;
 
 pub mod db_connection_pool;
 pub mod tables;
 
+pub use db_connection_pool::DbConnection;
 pub use db_connection_pool::DbConnectionPoolV2;
 
-pub type DbConnection = AsyncPgConnection;
-pub type DbConnectionPool = Pool<DbConnection>;
+pub type DieselConnection = AsyncPgConnection;
+pub type DbConnectionPool = Pool<DieselConnection>;
 
 /// Generic error type to forward errors from the database
 ///
 /// Useful for functions which only points of failure are the DB calls.
 #[derive(Debug, thiserror::Error)]
-#[error("an error occured while querying the database: {0}")]
-pub struct DatabaseError(#[from] diesel::result::Error);
+pub enum DatabaseError {
+    #[error("an error occured while querying the database: {0}")]
+    DieselError(#[from] diesel::result::Error),
+    #[error("an error occured while retrieving a connection from the pool: {0}")]
+    DatabasePoolError(#[from] DatabasePoolError),
+}
```

Comment:
I don't understand this change. There was two different types `DatabaseError` and `DatabasePoolError` that

1. serve different purposes, cf. their error messages
2. allows `editoast_models` to forward errors without exposing diesel's errors

Besides, `DatabasePoolError` also forwards `diesel::result::Error`. Its seems to me that the error types don't require any modification.